### PR TITLE
Tmp issue 186 b

### PR DESCRIPTION
**Pull request by diptighosalkar**
Graphical part for sigbuilder block parameter

### CODE REVIEW

File: webapp/Sigbuilder-graph.js [hunk 1]
```diff
@@ -0,0 +1,1872 @@
...
+        var diff_x = (((Math.abs(max_x_value_new - min_x_value_new))/100)*10).toFixed(2);
+        var diff_y = (((Math.abs(max_y_value_new - min_y_value_new))/100)*10).toFixed(2);
+        if(Math.abs(max_x - max_x_value_new) < parseFloat(diff_x)){
+            max_x = parseFloat(parseFloat(max_x) + parseFloat(diff_x)).toFixed(1);
+            drag_sig_chart.xAxis[0].update({
+                max: parseFloat(max_x)
+            });
+        }
+        if(Math.abs(max_y - max_y_value_new) < parseFloat(diff_y)){
+            max_y = parseFloat(parseFloat(max_y) + parseFloat(diff_y)).toFixed(1);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
            max_y = (max_y + diff_y).toFixed(1);
```

File: webapp/Sigbuilder-graph.js [hunk 2]
```diff
@@ -0,0 +1,1872 @@
...
+            drag_sig_chart.yAxis[0].update({
+                max: parseFloat(max_y)
+            });
+        }
+        //Added for negative/minimum value autoscale functionality for y axis
+        if(Math.abs(min_y - min_y_value_new) < parseFloat(diff_y)){
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
        if (diff_y > 0.1) {
```
is if correct? just check of diff_y > 0.1

File: webapp/Sigbuilder-graph.js [hunk 3]
```diff
@@ -0,0 +1,1872 @@
...
+            min_y = parseFloat(parseFloat(min_y) - parseFloat(diff_y)).toFixed(1);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
            min_y = (min_y - diff_y).toFixed(1);
```

File: webapp/Sigbuilder-graph.js [hunk 4]
```diff
@@ -0,0 +1,1872 @@
...
+        var x_max = parseFloat(document.getElementById("edit_xmax").value);
+        var y_min = parseFloat(document.getElementById("edit_ymin").value);
+        var y_max = parseFloat(document.getElementById("edit_ymax").value);
+        if(x_min < 0){
+            document.getElementById("messageLabel").innerHTML = "X should be positive";
+        }else{
+            document.getElementById("messageLabel").innerHTML = "";
+        }
+        sigbuilder_Graph.xAxis[0].update({
+                max: parseFloat(x_max)
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                max: x_max
```

parseFloat already called above

File: webapp/Sigbuilder-graph.js [hunk 5]
```diff
@@ -0,0 +1,1872 @@
...
+            });
+        sigbuilder_Graph.yAxis[0].update({
+                max: parseFloat(y_max)
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                max: y_max
```

File: webapp/Sigbuilder-graph.js [hunk 6]
```diff
@@ -0,0 +1,1872 @@
...
+            });
+        sigbuilder_Graph.yAxis[0].update({
+                min: parseFloat(y_min)
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                min: y_min
```

File: webapp/Sigbuilder-graph.js [hunk 7]
```diff
@@ -0,0 +1,1872 @@
...
+
+    myform.appendChild(ok_btn);
+    content.appendChild(myform);
+    var height = 220;
+    var wind = showModalWindow(graph, 'Scilab Multiple Values Request', content, 300, height);
+
+    var filename_input = document.getElementById("fileupload_0");
+    filename_input.readOnly = true;
+    filename_input.onclick = function(){
+        fileSelector.click();
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]: only show .dat and .txt files in file selector.

File: webapp/Sigbuilder-graph.js [hunk 8]
```diff
@@ -0,0 +1,1872 @@
...
+    for(var i = 0; i < labelArray.length; i++){
+        // Input Title
+        var namelabel = document.createElement('label');
+        namelabel.innerHTML = labelArray[i];
+        namelabel.style.marginLeft = "30px";
+        myform.appendChild(namelabel);
+
+        // Input
+        var input = document.createElement("input");
+        input.name = "edit_"+labelArray[i].toString();
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
        input.name = "edit_"+labelArray[i];
```
labelArray is a string array. don't use toString here.

File: webapp/Sigbuilder-graph.js [hunk 9]
```diff
@@ -0,0 +1,1872 @@
...
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+
+    // Line break
+    var linebreak = document.createElement('br');
+    myform.appendChild(linebreak);
+
+    var labelArray = ['Filename','Writing [C] format'];
+    var textValueArray = ["mydatafile.dat","%g %g"];
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]: you will have to use same file selector for saving as for loading.

File: webapp/Sigbuilder-graph.js [hunk 10]
```diff
@@ -0,0 +1,1872 @@
...
+            var ary = ["%"+firstchar,"%"+secondchar];
+            if((firstchar == 'g'||firstchar == 'f')&&(secondchar == 'g'||secondchar == 'f')){
+            if ("Blob" in window) {
+                var fileName = document.getElementById("filesave_0").value;
+                var textToWrite = "";
+                var x = 0;
+                var y = 0;
+                for (var i=0;i<pointlength;i++) {
+                    if(ary[0] == "%g"){
+                        if((sigbuilder_Graph.series[0].data[i].x).toString().includes(".")){
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                        var varx = sigbuilder_Graph.series[0].data[i].x;
                        if (varx.toString().includes(".")) {
```
line is too long. add a var first and use it everywhere.

File: webapp/Sigbuilder-graph.js [hunk 11]
```diff
@@ -0,0 +1,1872 @@
...
+            }else{
+                x_arr += x + ";";
+                y_arr += y + ";";
+            }
+        }
+        }else{
+            x_arr = "[0]";
+            y_arr = "[0]";
+        }
+        var sort_x_array = graphParameters.original_X_Points.slice(0);
```
[the remark this answers is not in the document]
Comment (diptighosalkar) [reply webapp/Sigbuilder-graph.js]: it is required as set y as [10,20,-30] again on parameter input box as when we enter x = [0,2,1] and y = [10,20,-30] as when chart is created it changes to x= [0,1,2] and y = [10,-30,20]

File: webapp/Sigbuilder-graph.js [hunk 12]
```diff
@@ -0,0 +1,1991 @@
...
+            if ("Blob" in window) {
+                var fileName = document.getElementById("filesave_0").value;
+                var textToWrite = "";
+                var x = 0;
+                var y = 0;
+                for (var i=0;i<pointlength;i++) {
+                    if(ary[0] == "%g"){
+                        var varx = sigbuilder_Graph.series[0].data[i].x;
+                        if (varx.toString().includes(".")) {
+                            x = parseFloat(sigbuilder_Graph.series[0].data[i].x).toFixed(6);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                            x = parseFloat(varx).toFixed(6);
```
use varx

File: webapp/Sigbuilder-graph.js [hunk 13]
```diff
@@ -0,0 +1,1991 @@
...
+                        }else{
+                            x = parseFloat(sigbuilder_Graph.series[0].data[i].x);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                            x = parseFloat(varx);
```

File: webapp/Sigbuilder-graph.js [hunk 14]
```diff
@@ -0,0 +1,1991 @@
...
+                        }
+                    }else{
+                        x = parseFloat(sigbuilder_Graph.series[0].data[i].x).toFixed(6);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                        x = parseFloat(varx).toFixed(6);
```

File: webapp/Sigbuilder-graph.js [hunk 15]
```diff
@@ -0,0 +1,1991 @@
...
+                    }
+                    if(ary[1] == "%g"){
+                        var vary = sigbuilder_Graph.series[0].data[i].y;
+                        if (vary.toString().includes(".")) {
+                            y = parseFloat(sigbuilder_Graph.series[0].data[i].y).toFixed(6);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                            y = parseFloat(vary).toFixed(6);
```
use vary

File: webapp/Sigbuilder-graph.js [hunk 16]
```diff
@@ -0,0 +1,1991 @@
...
+                        }else{
+                            y = parseFloat(sigbuilder_Graph.series[0].data[i].y);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                            y = parseFloat(vary);
```

File: webapp/Sigbuilder-graph.js [hunk 17]
```diff
@@ -0,0 +1,1991 @@
...
+                        }
+                    }else{
+                        y = parseFloat(sigbuilder_Graph.series[0].data[i].y).toFixed(6);
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]:
```suggestion
                        y = parseFloat(vary).toFixed(6);
```

File: webapp/Sigbuilder-graph.js [hunk 18]
```diff
@@ -0,0 +1,1991 @@
...
+        // Input Title
+        var namelabel = document.createElement('label');
+        namelabel.innerHTML = labelArray[i];
+        namelabel.style.marginLeft = "30px";
+        myform.appendChild(namelabel);
+
+        // Input
+        var input = document.createElement("input");
+        input.name = "edit_"+labelArray[i];
+        input.value = textValueArray[i];
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]: is it possible to use fileselector for i==0?

File: webapp/Sigbuilder-graph.js [hunk 19]
```diff
@@ -0,0 +1,1991 @@
...
```
Comment (sunilshetye) [inline webapp/Sigbuilder-graph.js]: is it possible to use file selector for i==0?

**Pull request merged**